core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -793,6 +804,9 @@ public Task getTask(String taskId) {
      * @throws ApplicationException If there was an error - caller should retry in this case.
      */
     public boolean decide(String workflowId) {
```

Comment:
@milongapau  Can you please tell me what is the code path taken when we see the race condition. 
From what I understand, A parallel execution of the same task can happen if the task is present multiple times in the task queue. How can this happen? 



I will be able to review the PR better if I can understand this and also would be helpful to me as I am thinking of picking up the implementation of your locking service.

Reply:
@naveenchlsn see the example in analysis section of documentation, the issue is a workflow can be executed more than once resulting in inconsistency.